### PR TITLE
fix(theme-classic): improve aria label of color mode toggle

### LINKED ISSUE #6665: a11y: light & dark mode checkbox ambiguous on current state ("checked")
### Have you read the Contributing Guidelines on issues?

- [X] I have read the [Contributing Guidelines on issues](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#reporting-new-issues).

### Prerequisites

- [X] I'm using the latest version of Docusaurus.
- [X] I have tried the `npm run clear` or `yarn clear` command.
- [X] I have tried `rm -rf node_modules yarn.lock package-lock.json` and re-installing packages.
- [X] I have tried creating a repro with https://new.docusaurus.io.
- [X] I have read the console error message carefully (if applicable).

### Description

Right now the 🌞/🌜 light/dark mode toggle checkbox has the `aria-label` `"Switch between dark and light mode"`. "Checked" and "not checked" are ambiguous terms without more context because some websites default to dark mode and/or don't respect system defaults.

I realize it might seem at first like the group of people who want a screenreader _and_ care about light vs dark mode might seem small. A couple quick scenarios that come to mind:

* Some folks such as low vision or with reading difficulties use screenreaders to assist with difficult-to-understand pages
* Some folks can't see the website in one mode, and can only see it in the other

### Steps to reproduce

1. Open a screenreader such as NVDA on Windows or VoiceOver on Mac
2. Put screenreader/text focus on the light/dark mode toggle

### Expected behavior

I'm not an expert here, but maybe a message like...

```
switch from light to dark mode checkbox, not checked
```

...or maybe use a button instead of a checkbox behind the scenes?

```
switch between light and dark mode checkbox (current value light)
```

### Actual behavior

The narration by default in light mode is roughly:

```
switch between light and dark mode checkbox, not checked
```

### Your environment

- Public source code: https://github.com/typescript-eslint/typescript-eslint
- Public site URL: https://typescript-eslint.io
- Docusaurus version used: 2.0.0-beta.15
- Environment name and version (e.g. Chrome 89, Node.js 16.4): Chrome 89
- Operating system and version (e.g. Ubuntu 20.04.2 LTS): Windows 10

### Reproducible demo

https://typescript-eslint.io

### Self-service

- [X] I'd be willing to fix this bug myself.

### PR DESCRIPTION
**Pull request by Josh-Cena**
## Motivation

Close #6665.

### Have you read the [Contributing Guidelines on pull requests](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#pull-requests)?

Yes

### CODE REVIEW

**Comment by Josh-Cena** [comment]
@JoshuaKGoldberg The GitHub review requesting panel is pretty awkward and I can't find your name. If you can take a look and see if your screenreader reads something better, it would be great.

File: packages/docusaurus-theme-classic/src/theme/Toggle/index.tsx [hunk 1]
```diff
@@ -62,11 +62,26 @@ const ToggleComponent = memo(
...
+            {
+              mode: checked
+                ? translate({
+                    message: 'dark mode',
+                    id: 'theme.colorToggle.ariaLabel.mode.dark',
+                    description: 'The name for the dark color mode',
+                  })
+                : translate({
+                    message: 'light mode',
+                    id: 'theme.colorToggle.ariaLabel,mode.light',
```
Comment (duanwilliam) [inline packages/docusaurus-theme-classic/src/theme/Toggle/index.tsx:80]: should this be `.` not `,`?

Comment (Josh-Cena) [reply packages/docusaurus-theme-classic/src/theme/Toggle/index.tsx:80]: Oops, good catch!

**Review by lex111 — APPROVED** [review]
:+1:

**Review by JoshuaKGoldberg — COMMENTED** [review]
Yup, I'm not an expert but in general looks good to me, thank you!

**Pull request merged**